### PR TITLE
Fix required arguments output

### PR DESCRIPTION
**Pull request by emazzotta**
This PR fixes this issue https://github.com/nodeca/argparse/issues/120
The change was very easy to make, thanks for structuring your code well! 👍 

Please run `npm publish` afterwards, I already bumped the version.

Some explanations:
* Why did I add a yarn.lock? it's good for deterministic downloading of dependencies (alternatively, npm > 5.x.x can create a package-lock.json)
* Why did I rename the 'Optional arguments' and 'Positional arguments' to 'optional arguments' and 'positional arguments' -> That's how it's done in the Python argparse lib

### CODE REVIEW

**Comment by emazzotta** [comment]
`{ optional: false }` doesn't work for me either, it will still be shown below the `Optional arguments:` output. Also, Python argparse uses `required=True/False` for this, and as I understand it, this package tries to resemble its Python counterpart.

**Comment by puzrin** [comment]
Please, remove version bump & yarn, don't mix completely different things to simgle PR.

Do i understand right, that `Action.prototype.isRequired` exists in python's mainstream and was missed here?

**Comment by emazzotta** [comment]
@puzrin Sure, I removed the version bump and yarn.

Not sure if argparse has this exact method, but this was definitely missing and exists for Python's argparse:
`this._requireds = this.addArgumentGroup({ title: 'required arguments' });`

File: lib/action.js [hunk 1]
```diff
@@ -108,6 +108,15 @@ Action.prototype.getName = function () {
 };
 
 /**
+ * Action#isRequired -> Boolean
+ *
+ * Return true if required
+ **/
+Action.prototype.isRequired = function () {
```
Comment (puzrin) [inline lib/action.js]: I see something similar in mainstream https://github.com/python/cpython/blob/2bd37c227e8a042e036c7455d974e3d0b36aed53/Lib/argparse.py#L817.

Seems ok.

File: lib/argument_parser.js [hunk 2]
```diff
@@ -78,8 +78,9 @@ function ArgumentParser(options) {
 
   this.formatterClass = (options.formatterClass || HelpFormatter);
   this.fromfilePrefixChars = options.fromfilePrefixChars || null;
-  this._positionals = this.addArgumentGroup({ title: 'Positional arguments' });
-  this._optionals = this.addArgumentGroup({ title: 'Optional arguments' });
+  this._positionals = this.addArgumentGroup({ title: 'positional arguments' });
+  this._requireds = this.addArgumentGroup({ title: 'required arguments' });
```
Comment (puzrin) [inline lib/argument_parser.js]: https://github.com/python/cpython/blob/2bd37c227e8a042e036c7455d974e3d0b36aed53/Lib/argparse.py#L1649 - can't find appropriate line in mainstream

File: lib/argument_parser.js [hunk 3]
```diff
@@ -197,7 +198,9 @@ ArgumentParser.prototype.addSubparsers = function (options) {
 };
 
 ArgumentParser.prototype._addAction = function (action) {
-  if (action.isOptional()) {
+  if (action.isRequired()) {
```
Comment (puzrin) [inline lib/argument_parser.js]: https://github.com/python/cpython/blob/2bd37c227e8a042e036c7455d974e3d0b36aed53/Lib/argparse.py#L1725

also don't see anything similar in mainstream

**Comment by emazzotta** [comment]
@puzrin you're right, I was looking at the wrong thing, I can actually just add this in my own project:

```javascript
const requiredArgs = parser.addArgumentGroup({ title: 'required arguments' });

requiredArgs.addArgument(
  ['-u', '--urls'], {
    action: 'store',
    required: true,
    nargs: Const.ONE_OR_MORE,
    help: 'Bla',
  },
);
```

So except for lowercasing Positional and Optional, this can be closed. 
I'll raise a new PR, sorry for the inconvenience.

**Pull request closed without merging**